build/page-description.ts

```diff
@@ -0,0 +1,83 @@
+import { DEFAULT_LOCALE } from "../libs/constants/index.js";
+import { Doc } from "../libs/types/document.js";
+
+export function getPageDescription(
+  doc: Partial<Doc>,
+  pageType: string
+): string | undefined {
+  if (doc.locale === DEFAULT_LOCALE) {
+    const sections = doc.toc?.map(({ text }) => text) ?? [];
+
+    const syntaxItems = Object.entries({
+      Value: "type",
+      "Event type": "type",
+      Syntax: "syntax",
+      Parameters: "parameters",
+      Constructor: "constructor",
+      "Instance properties": "properties",
+      "Event properties": "properties",
+      "Instance methods": "methods",
+      "Return value": "return value",
+    })
+      .filter(([section]) => sections.includes(section))
+      .map(([, text]) => text);
+
+    const otherItems = Object.entries({
+      Exceptions: "exceptions",
+      Examples: "code examples",
+      Specifications: "specifications",
+      "Browser compatibility": "browser compatibility",
+    })
+      .filter(([section]) => sections.includes(section))
+      .map(([, text]) => text);
+
+    const syntaxContents = enumerate(...syntaxItems);
+    const otherContents = enumerate(...otherItems);
+    const contents = [
+      syntaxContents ? `its ${syntaxContents}` : "",
+      otherContents,
+    ]
+      .filter(Boolean)
+      .join(", ");
+
+    switch (pageType) {
+      case "web-api-instance-property":
+      case "web-api-static-property":
+        // "Learn about the Window.localStorage property, ..."
+        // "Learn about the static Notification.permission property, ..."
+        return `Learn about the ${doc.title.replace(/^(.*?): (.*?) (static )?property$/, "$3$1.$2 property")}, including ${contents}.`;
```

Comment:
This is a double-edged sword. If the pages with bad summaries are fewer than the pages with good summaries, we'll end up harming the search rankings of most of the pages.

How about we augment the existing summary instead?
```plain
The Notification.permission is a static property. The permission read-only static property of the Notification interface indicates the current permission granted by the user for the current ...

The Window.localStorage is an instance property.  The localStorage read-only property of the window interface allows you to access a Storage object for the Document's origin; the stored data ...
```
Above can be achieved using:
```suggestion
      case "web-api-instance-property":
        return `The ${doc.title.replace(/^(.*?): (.*?) property$/, "$3$1.$2")} is an instance property. ${doc.summary}`;
      case "web-api-static-property":
        return `The ${doc.title.replace(/^(.*?): (.*?) static property$/, "$3$1.$2")} is a static property. ${doc.summary}`;
```

This will get us in the FAQ slot as well:
![1](https://github.com/user-attachments/assets/c7c5341c-2af1-43ab-9b3b-03d6d1629200)